src/lib/Types/Core.hs

```diff
@@ -2156,3 +2157,6 @@ instance SubstE AtomSubstVal (EffectP AtomName) where
       RWSEffect rws v'
     ExceptionEffect -> ExceptionEffect
     IOEffect        -> IOEffect
+    UserEffect v    ->
+      let (Rename v') = env ! v
```

Comment:
Is there a reason why the other cases are impossible?  Should that be checked at runtime?

Reply:
The type system determined the other alternatives are impossible thanks to the name color.

Reply:
Maybe note that with a comment, for the benefit of the unwary?